### PR TITLE
Comment and Refactor a few files

### PR DESCRIPTION
**Pull request by tedinski**
### Description of changes: 

1. Add a lot of doc comments and other comments, focusing mostly on `statement.rs`
2. Move some code around.
3. Rename some bad variables.

Please note: anything that looks like new/deleted code is actually just moving. Github makes it a bit rough to review these diffs, I think.

### Testing:

* How is this change tested?

* Is this a refactor change?

### Checklist
- [ ] Each commit message has a non-empty body, explaining why the change was made
- [ ] Methods or procedures are documented
- [ ] Regression or unit tests are included, or existing tests cover the modified code
- [ ] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: kani-compiler/src/codegen_cprover_gotoc/codegen/assert.rs [hunk 1]
```diff
@@ -66,6 +71,7 @@ impl<'tcx> GotocCtx<'tcx> {
         )
     }
 
+    /// A shorthand for generating a CBMC assert-false. TODO: This should probably be eliminated!
```
Comment (adpaco-aws) [inline kani-compiler/src/codegen_cprover_gotoc/codegen/assert.rs:74]: Do we want an issue for this?

File: kani-compiler/src/codegen_cprover_gotoc/codegen/block.rs [hunk 2]
```diff
@@ -1,12 +1,16 @@
 // Copyright Kani Contributors
 // SPDX-License-Identifier: Apache-2.0 OR MIT
 
-//! This file contains functions related to codegenning MIR blocks into gotoc
```
Comment (adpaco-aws) [inline kani-compiler/src/codegen_cprover_gotoc/codegen/block.rs:4]: Why remove this? I feel like having at least one comment is better than not having it at all.

File: kani-compiler/src/codegen_cprover_gotoc/codegen/statement.rs [hunk 3]
```diff
@@ -24,19 +23,166 @@ use rustc_target::abi::{FieldsShape, Primitive, TagEncoding, Variants};
...
-            FN_RETURN_VOID_VAR_NAME,
-            is_file_local,
-            ty,
-            Location::none(),
-            |_, _| None,
-        );
-        Stmt::ret(Some(var), Location::none())
+    /// Generate Goto-C for MIR [Statement]s.
+    /// This does not cover all possible "statements" because MIR distinguishes between ordinary
+    /// statements and [Terminator]s, which can exclusively appear the the end of a basic block.
```
Comment (adpaco-aws) [inline kani-compiler/src/codegen_cprover_gotoc/codegen/statement.rs:28]:
```suggestion
    /// statements and [Terminator]s, which can exclusively appear at the end of a basic block.
```

**Review by adpaco-aws — APPROVED** [review]
Thanks! Just a few comments.

File: kani-compiler/src/codegen_cprover_gotoc/codegen/assert.rs (hunk 1 above)
Comment (tedinski) [reply kani-compiler/src/codegen_cprover_gotoc/codegen/assert.rs:74]: It's the assertions audit. I will be doing it next. :)

File: kani-compiler/src/codegen_cprover_gotoc/codegen/block.rs (hunk 2 above)
Comment (tedinski) [reply kani-compiler/src/codegen_cprover_gotoc/codegen/block.rs:4]: I basically just moved the comment from the file to the function.

Comment (adpaco-aws) [reply kani-compiler/src/codegen_cprover_gotoc/codegen/block.rs:4]: Right, I thought this implementation was way longer.

**Pull request merged**